data-prepper-plugins/s3-source/src/test/java/org/opensearch/dataprepper/plugins/source/configuration/AwsAuthenticationOptionsTest.java

```diff
@@ -116,6 +138,84 @@ void authenticateAWSConfiguration_should_return_s3Client_with_sts_role_arn_when_
+
+            reflectivelySetField(awsAuthenticationOptions, "awsRegion", "us-east-1");
+            reflectivelySetField(awsAuthenticationOptions, "awsStsRoleArn", "arn:aws:iam::123456789012:iam-role");
+            reflectivelySetField(awsAuthenticationOptions, "awsStsHeaderOverrides", overrideHeaders);
+
+            when(stsClientBuilder.region(Region.US_EAST_1)).thenReturn(stsClientBuilder);
+
+            final AssumeRoleRequest.Builder assumeRoleRequestBuilder = mock(AssumeRoleRequest.Builder.class);
+            when(assumeRoleRequestBuilder.roleSessionName(anyString()))
+                    .thenReturn(assumeRoleRequestBuilder);
+            when(assumeRoleRequestBuilder.roleArn(anyString()))
+                    .thenReturn(assumeRoleRequestBuilder);
+            when(assumeRoleRequestBuilder.overrideConfiguration(any(Consumer.class)))
+                    .thenReturn(assumeRoleRequestBuilder);
+
+            final AwsCredentialsProvider actualCredentialsProvider;
+            try (final MockedStatic<StsClient> stsClientMockedStatic = mockStatic(StsClient.class);
+                 final MockedStatic<AssumeRoleRequest> assumeRoleRequestMockedStatic = mockStatic(AssumeRoleRequest.class)) {
+                stsClientMockedStatic.when(StsClient::builder).thenReturn(stsClientBuilder);
+                assumeRoleRequestMockedStatic.when(AssumeRoleRequest::builder).thenReturn(assumeRoleRequestBuilder);
+                actualCredentialsProvider = awsAuthenticationOptions.authenticateAwsConfiguration();
+            }
+
+            assertThat(actualCredentialsProvider, instanceOf(AwsCredentialsProvider.class));
+
+            final ArgumentCaptor<Consumer<AwsRequestOverrideConfiguration.Builder>> configurationCaptor = ArgumentCaptor.forClass(Consumer.class);
+
+            verify(assumeRoleRequestBuilder).roleArn("arn:aws:iam::123456789012:iam-role");
+            verify(assumeRoleRequestBuilder).roleSessionName(anyString());
+            verify(assumeRoleRequestBuilder).overrideConfiguration(configurationCaptor.capture());
+            verify(assumeRoleRequestBuilder).build();
+            verifyNoMoreInteractions(assumeRoleRequestBuilder);
+
+            final Consumer<AwsRequestOverrideConfiguration.Builder> actualOverride = configurationCaptor.getValue();
+
+            final AwsRequestOverrideConfiguration.Builder configurationBuilder = mock(AwsRequestOverrideConfiguration.Builder.class);
+            actualOverride.accept(configurationBuilder);
+            verify(configurationBuilder).putHeader(headerName1, headerValue1);
+            verify(configurationBuilder).putHeader(headerName2, headerValue2);
+            verifyNoMoreInteractions(configurationBuilder);
+        }
+
+        @Test
+        void authenticateAWSConfiguration_should_override_STS_Headers_when_HeaderOverrides_are_empty() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
minor: I think the test name should be `authenticateAWSConfiguration_should_not_override_STS_Headers_when_HeaderOverrides_are_empty` as it doesn't override if it's empty